lib/api/src/externals/function.rs

```diff
@@ -714,9 +794,10 @@ pub(crate) trait VMDynamicFunction {
     fn function_type(&self) -> &FunctionType;
 }
 
+#[derive(Clone)]
 pub(crate) struct VMDynamicFunctionWithoutEnv {
     #[allow(clippy::type_complexity)]
-    func: Box<dyn Fn(&[Val]) -> Result<Vec<Val>, RuntimeError> + 'static>,
+    func: Arc<dyn Fn(&[Val]) -> Result<Vec<Val>, RuntimeError> + 'static>,
```

Comment:
I'm surprised to see `Arc` in a VM struct. Either this shouldn't be a VM struct because the compiler doesn't need to read its contents, or the Arc is actually an FFI safe type, or maybe this one field of this struct isn't used by the compiler and that's OK and we should put a comment on it indicating as much.

Reply:
So this function has the name "VM" but it's actually in the API crate.

The only way this is interacted with at the compiler / vm levels is via `Clone` I believe (and perhaps passing a pointer to the `VMDynamicFunctionWithoutEnv` type itself around)

Reply:
@syrusakbary thoughts on the names of these types? I agree with Nick that having types that start with `VM` in API is a bit confusing to start. Also as Nick said, types that start with `VM` that aren't meant for low level access seems bad too.

Reply:
Agreed, perhaps the `VM` part of the name should be removed